### ISSUE TITLE
[@typescript-eslint/yoda] New rule proposal

### ISSUE THREAD

**Issue by krisztianb** [typescript-eslint/typescript-eslint#3596]
ESLint has a [yoda](https://eslint.org/docs/rules/yoda) rule which reports the following code as problematic:

```ts
if (4 === variable) {
    // do something
}
```

However that rule has no type information and therefore cannot check a similar case:

```ts
enum MagicNumber {
    One = 1,
    Two = 2,
    Three = 3,
    Four = 4,
}

if (MagicNumber.Four === variable) {
    // do something
}
```

It would be useful to have a similar rule for TypeScript that can deal with enumerations too.

What do you think?

**Comment by bradzacher** [comment]
in my **opinion**, yoda is not a good lint rule. I don't think the convention is a good one.
From what I can tell with a [quick github search](https://github.com/search?q=yoda+filename%3A.eslintrc+filename%3A.eslintrc.json+filename%3A.eslintrc.js&type=Code), it's barely used as well.

I'll leave this open for now, but I personally don't think it's worth the effort.

**Comment by JoshuaKGoldberg** [comment]
This has been open for over 3 years and has only had six positive community interactions. That's less than one person every six months. Given ESLint's [general deprioritization of stylistic rules](https://eslint.org/blog/2020/05/changes-to-rules-policies/#what-are-the-changes), I'm going to go ahead and close this one out. Cheers!
